src/main/java/gov/adlnet/xapi/client/StatementClient.java

```diff
@@ -200,9 +200,7 @@ private StatementClient addFilter(String key, String value) {
 				client.filters = new TreeMap<String, String>();
 			}
 			if (this.filters != null) {
-				for (Entry<String, String> filter : filters.entrySet()) {
-					client.filters.put(filter.getKey(), filter.getValue());
-				}
+				client.filters.putAll(this.filters);
```

Comment:
That would be ok...just add a comment to it explaining why it's public instead of private.